framework-plugins/lisk-framework-forger-plugin/src/forger_plugin.ts

```diff
@@ -122,11 +157,159 @@ export class ForgerPlugin extends BasePlugin {
 
 		return new KVStore(dirPath);
 	}
+
+	private _subscribeToChannel(): void {
+		// eslint-disable-next-line @typescript-eslint/no-misused-promises
+		this._channel.subscribe('app:block:new', async (newBlock: EventInfoObject) => {
+			const { block } = newBlock.data as Data;
+			await this._incrementForgerInfo(block);
+		});
+
+		// eslint-disable-next-line @typescript-eslint/no-misused-promises
+		this._channel.subscribe('app:block:delete', async (deleteBlock: EventInfoObject) => {
+			const { block } = deleteBlock.data as Data;
+			await this._decrementForgerInfo(block);
+		});
+	}
+
+	private async _setForgersList(): Promise<void> {
+		this._forgersList = await this._channel.invoke<Forger[]>('app:getForgingStatusOfAllDelegates');
+	}
+
+	private async _getForgerInfo(forgerAddress: string): Promise<ForgerInfo> {
+		let forgerInfo;
+		try {
+			forgerInfo = await this._forgerPluginDB.get(`${DB_KEY_FORGER_INFO}:${forgerAddress}`);
+		} catch (error) {
+			debug(`Forger info does not exists for delegate: ${forgerAddress}`);
+			return {
+				totalProducedBlocks: 0,
+				totalReceivedFees: BigInt(0),
+				totalReceivedRewards: BigInt(0),
+				votesReceived: [],
+			};
+		}
+
+		return codec.decode<ForgerInfo>(forgerInfoSchema, forgerInfo);
+	}
+
+	private async _setForgerInfo(forgerAddress: string, forgerInfo: ForgerInfo): Promise<void> {
+		const encodedForgerInfo = codec.encode(forgerInfoSchema, forgerInfo);
+		await this._forgerPluginDB.put(`${DB_KEY_FORGER_INFO}:${forgerAddress}`, encodedForgerInfo);
+	}
+
+	private _getGeneratorAndPayloadInfo(block: string): GeneratorPayloadInfo {
+		const {
+			header: { generatorPublicKey, reward },
+			payload,
+		} = this.codec.decodeBlock(block);
+		const forgerAddress = getAddressFromPublicKey(
+			Buffer.from(generatorPublicKey, 'base64'),
+		).toString('base64');
+		const forgerAddressBuffer = Buffer.from(forgerAddress, 'base64');
+
+		return {
+			forgerAddress,
+			forgerAddressBuffer,
+			reward,
+			payload,
+		};
+	}
+
+	private async _incrementForgerInfo(block: string): Promise<void> {
+		const { forgerAddress, reward, payload } = this._getGeneratorAndPayloadInfo(block);
+		const forgerInfo = await this._getForgerInfo(forgerAddress);
+		let isUpdated = false;
+
+		if (this._forgersList.find(forger => forger.address === forgerAddress)) {
+			forgerInfo.totalProducedBlocks += 1;
+			forgerInfo.totalReceivedRewards += BigInt(reward);
+			forgerInfo.totalReceivedFees += await this._getFee(payload, block);
+			isUpdated = true;
+		}
+
+		for (const trx of payload) {
+			if (trx.type === VoteTransaction.TYPE) {
+				for (const vote of (trx.asset as Asset).votes) {
+					const registeredDelegateIndex = this._forgersList.findIndex(
+						forger => forger.address === vote.delegateAddress,
+					);
+					if (registeredDelegateIndex !== -1) {
+						const delegateVoteIndex = forgerInfo.votesReceived.findIndex(aVote =>
+							aVote.address.equals(Buffer.from(vote.delegateAddress, 'base64')),
```

Comment:
is `aVote.address` Buffer?

Reply:
yes, its the forgerInfo address.. decoded as buffer.